client/client_pool/src/external_client.cpp

```diff
@@ -235,21 +235,28 @@ std::pair<int32_t, ConcordClient::PendingReplies> ConcordClient::SendPendingRequ
 }
 
 BaseCommConfig* ConcordClient::CreateCommConfig() const {
-  const auto& client_conf = pool_config_.participant_nodes.at(0).externalClients.at(client_id_);
   const auto commType =
       pool_config_.comm_to_use == "tls"
           ? TlsTcp
           : pool_config_.comm_to_use == "tcp" ? PlainTcp : pool_config_.comm_to_use == "udp" ? PlainUdp : SimpleAuthTcp;
+#if defined(USE_COMM_PLAIN_TCP) || defined(USE_COM_TLS_TCP)
+  const auto& client_conf = pool_config_.participant_nodes.at(0).externalClients.at(client_id_);
   const auto listenPort = client_conf.client_port;
   const auto bufferLength = std::stoul(pool_config_.concord_bft_communication_buffer_length);
   const auto selfId = client_conf.principal_id;
-  if (commType == PlainTcp)
+#endif
+  if (commType == PlainTcp) {
+#ifdef USE_COMM_PLAIN_TCP
     return new PlainTcpConfig{"external_client",
                               listenPort,
                               static_cast<uint32_t>(bufferLength),
                               pool_config_.replicas,
                               static_cast<int32_t>(pool_config_.num_replicas - 1),
                               selfId};
+#endif
+    throw std::runtime_error("FOR PLAIN TCP COMM SHOULD HAVE BEEN COMPILED WITH \"USE_COMM_PLAIN_TCP\"");
```

Comment:
Why not wrap the whole runtime condition with an #ifdef?
Then you won't need to throw.

Reply:
because the choice of communication is during run-time but classes could be ifdef'ed out

Reply:
you did once you added the throw